src/app/scripts/cac/control/cac-control-filter-options.js

```diff
@@ -30,26 +30,46 @@ CAC.Control.FilterOptions = (function ($) {
         events: events,
         eventNames: eventNames,
         getFilter: getFilter,
-        setFilter: setFilter
+        setFilter: setFilter,
+        destroy: destroy
     };
 
     return FilterOptionsControl;
 
-    function initialize() {
-        // update classes on filter toggle buttons
-        $(options.selectors.filterToggle).on('click', options.selectors.filterOption, function(e) {
-            e.preventDefault();
-
-            $(this).addClass(options.selectors.onClass)
+    // helper to set the 'on' class for the selected option, and unset it for the others
+    function toggleOn(selector) {
+        $(selector).addClass(options.selectors.onClass)
                 .removeClass(options.selectors.offClass)
                 .siblings(options.selectors.filterOption)
                     .removeClass(options.selectors.onClass)
                     .addClass(options.selectors.offClass);
+    }
+
+    function initialize() {
+        // filter toggle button row event handler
+        $(options.selectors.filterToggle).on('click', options.selectors.filterOption, function(e) {
+            e.preventDefault();
+            toggleOn(this);
+            events.trigger(eventNames.toggle, getFilter());
+        });
+
+        // filter drop-down button event handler
+        $(options.selectors.filterToggle).on('change', function(e) {
+            e.preventDefault();
+            if (!e.target.selectedOptions || ! e.target.selectedOptions.length) {
```

Comment:
Space after `!` looks odd, especially right after one with no space.

Reply:
Oops, that was accidental. Good catch.